### PR TITLE
Remove type field from internal PutMappingRequest

### PR DESCRIPTION
**Pull request by romseygeek**
External API requests can no longer include types, so we have no need to pass this
information along in internal PutMappingClusterStateUpdateRequest objects, or on
PutMappingRequests used by the internal node client.

Relates to #41059

### CODE REVIEW

**Comment by romseygeek** [comment]
There are a number of changes here to just pass `_doc` to things like mapper parsers; these will be removed later on, but I'm trying to keep individual PRs as small as possible.

**Review by mayya-sharipova — APPROVED** [review]
@romseygeek Thanks Alan! I left small comments, but overall LGTM

File: server/src/main/java/org/elasticsearch/action/admin/indices/mapping/put/PutMappingRequest.java [hunk 1]
```diff
@@ -80,7 +80,9 @@ public PutMappingRequest(StreamInput in) throws IOException {
         super(in);
         indices = in.readStringArray();
         indicesOptions = IndicesOptions.readIndicesOptions(in);
-        type = in.readOptionalString();
+        if (in.getVersion().before(Version.V_8_0_0)) {
+            in.readOptionalString();
```
Comment (jtibshirani) [inline server/src/main/java/org/elasticsearch/action/admin/indices/mapping/put/PutMappingRequest.java]: We could check the type and throw an `IllegalArgumentException` here as we've been doing in other places.

File: server/src/test/java/org/elasticsearch/index/mapper/UpdateMappingTests.java [hunk 2]
```diff
@@ -200,7 +200,6 @@ public void testMappingVersion() {
             final long previousVersion = clusterService.state().metaData().index("test").getMappingVersion();
             final PutMappingRequest request = new PutMappingRequest();
             request.indices("test");
-            request.type("type");
```
[the remark this answers is not in the document]
Comment (romseygeek) [reply server/src/test/java/org/elasticsearch/index/mapper/UpdateMappingTests.java]: I plan on doing this in subsequent PRs - for example, line 167 will need to change when we remove `type` from `MapperService.merge`

File: server/src/test/java/org/elasticsearch/action/admin/indices/mapping/put/PutMappingRequestTests.java [hunk 3]
```diff
@@ -133,7 +126,6 @@ private static PutMappingRequest createTestItem() throws IOException {
         PutMappingRequest request = new PutMappingRequest(index);
 
         String type = randomAlphaOfLength(5);
-        request.type(type);
```
[the remark this answers is not in the document]
Comment (romseygeek) [reply server/src/test/java/org/elasticsearch/action/admin/indices/mapping/put/PutMappingRequestTests.java]: ++

File: server/src/main/java/org/elasticsearch/action/admin/indices/mapping/put/PutMappingRequest.java (hunk 1 above)
Comment (romseygeek) [reply server/src/main/java/org/elasticsearch/action/admin/indices/mapping/put/PutMappingRequest.java]: ++

File: server/src/main/java/org/elasticsearch/action/admin/indices/mapping/put/PutMappingClusterStateUpdateRequest.java [hunk 4]
```diff
@@ -26,29 +26,18 @@
...
     public String source() {
         return source;
     }
 
-    public PutMappingClusterStateUpdateRequest source(String source) {
-        this.source = source;
-        return this;
+    @Override
+    public String toString() {
+        return "put-mapping";
```
[the remark this answers is not in the document]
Comment (romseygeek) [reply server/src/main/java/org/elasticsearch/action/admin/indices/mapping/put/PutMappingClusterStateUpdateRequest.java]: Looking at the history, I'm not sure why I added this in the first place, as it didn't have a `toString()` previously.  I'll remove it.

**Comment by romseygeek** [comment]
@elasticmachine update branch

**Pull request merged**